### PR TITLE
ISSUE-536 support CompoundIdentifier

### LINKED ISSUE #536: Unsupported expression: a.number, type: CompoundIdentifier
**Describe the bug**

client:
```
mysql> SELECT a.number FROM numbers(3) AS a;
ERROR 1105 (HY000): Code: 5, displayText = Unsupported expression: a.number, type: CompoundIdentifier([Ident { value: "a", quote_style: None }, Ident { value: "number", quote_style: None }]).
```

server log:
```
ERROR fuse_query::servers::mysql::endpoints::endpoint_on_query] OnQuery Error: Code: 5, displayText = "Unsupported expression: a.number, type: CompoundIdentifier([Ident { value: \"a\", quote_style: None }, Ident { value: \"number\", quote_style: None }])".
    
       0: common_exception::exception::ErrorCodes::SyntexException
                 at common/exception/src/exception.rs:38:41
          fuse_query::sql::plan_parser::PlanParser::sql_to_rex
                 at fusequery/query/src/sql/plan_parser.rs:489:34
       1: fuse_query::sql::plan_parser::PlanParser::sql_select_to_rex
                 at fusequery/query/src/sql/plan_parser.rs:256:62
       2: fuse_query::sql::plan_parser::PlanParser::select_to_plan::{{closure}}
                 at fusequery/query/src/sql/plan_parser.rs:203:22
          core::iter::adapters::map::map_try_fold::{{closure}}

```

### PR DESCRIPTION
**Pull request by jyizheng**
## Summary

Support   CompoundIdentifier for sql like:

SELECT a.number FROM numbers(3) AS a;

## Changelog

- New Feature

## Related Issues

Fixes #536 

## Test Plan
Stateless Tests

### CODE REVIEW

File: tests/suites/0_stateless/00_0000_dummy_select_1.sql [hunk 1]
```diff
@@ -7,3 +7,4 @@ SELECT NOT(1=1) from numbers(3);
 SELECT TRUE;
 SELECT FALSE;
 SELECT NOT(TRUE);
+SELECT a.number FROM numbers(3) AS a order by a.number;
```
Comment (zhang2014) [inline tests/suites/0_stateless/00_0000_dummy_select_1.sql:10]: What will happen?
``` sql
SELECT b.number FROM numbers(3) AS a ORDER BY a.number;
SELECT a.number FROM numbers(3) AS a ORDER BY b.number;
SELECT b.number FROM (SELECT * FROM numbers(3) AS b ORDER BY a.number) AS a;
SELECT b.number FROM (SELECT * FROM numbers(3) AS b) AS a ORDER BY b.number;
```

Comment (jyizheng) [reply tests/suites/0_stateless/00_0000_dummy_select_1.sql:10]: All above four statement has similar issue like:

`ERROR 1105 (HY000): Code: 25, displayText = Unknown Table '"b"'.`

Comment (zhang2014) [reply tests/suites/0_stateless/00_0000_dummy_select_1.sql:10]: 👍 Maybe we add them to the test?

Comment (jyizheng) [reply tests/suites/0_stateless/00_0000_dummy_select_1.sql:10]: Added.  PTAL @zhang2014

**Review by zhang2014 — APPROVED** [review]
LGTM

**Pull request merged**